src/components/ArticlesList.jsx

```diff
@@ -2,30 +2,89 @@ import { fetchArticles } from "../api.js";
 import Articles from "./Articles.jsx";
 import { useState, useEffect } from "react";
 import Loading from "./Loading.jsx";
-import {useSearchParams} from "react-router-dom";
+import { useSearchParams } from "react-router-dom";
 
 const ArticlesList = () => {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [sortAttributes, setSortAttributes] = useState([]);
+  const [order, setOrder] = useState(true);
   const [searchParams, setSearchParams] = useSearchParams();
-  const filterByTopic = searchParams.get('topic');
+  const filterByTopic = searchParams.get("topic");
+  const sortBy = searchParams.get("sort_by");
+  const orderBy = searchParams.get("order");
 
+  const setOrderParams = (order) => {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("order", order);
+    setSearchParams(newParams);
+  };
+
+  const setSortByParams = (sortBy) => {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("sort_by", sortBy);
+    setSearchParams(newParams);
+  };
```

Comment:
Just wondering if this can be dried up at all seeing as they're doing almost the same thing